exs/mundo_2/ruby/058.rb

```diff
@@ -0,0 +1,29 @@
+=begin
+Desafio 058
+
+Problema: Melhore o jogo do desafio 028 onde o computador vai "pensar" em um
+          número entre 0 e 10. Só que agora o jogador vai tentar adivinhar até
+          acertar, mostrando no final quantos palpites foram necessários para vencer.
+
+Resolução do problemas:
+=end
+
+pc = rand(1..10)
+chutes = 0
+acertou = false
+
+until acertou
+	print"Qual o seu palpite? "
+	n = gets.chomp.to_i
+	if n == pc
+		acertou = true
+	elsif n < pc
+		puts"mais que isso!"
+		chutes += 1
+	elsif n > pc
+		puts"menos que isso!"
+		chutes += 1
+	end
+end
```

Comment:
Mesmo que o usuário acerte ou erro, ela terá feito um palpite. Sendo assim, você sempre acrescentará 1 nos chutes.
```suggestion
	if n == pc
		acertou = true
	elsif n < pc
		puts"mais que isso!"
	elsif n > pc
		puts"menos que isso!"
	end
	chutes += 1
end
```